test/accumulation.jl

```diff
@@ -0,0 +1,62 @@
+@testset "accumulation.jl" begin
+    @testset "scalar" begin
+        @test 16 == add!!(12, 4)
+    end
+
+    @testset "Differentials" begin
+        @test 16 == add!!(12, @thunk(2*2))
+        @test 16 == add!!(16, Zero())
+
+        @test 16 == add!!(16, DoesNotExist())  # Should this be an error?
```

Comment:
do we have/need an issue for this?

Reply:
we don't, I am not sure we need one yet.